### PR TITLE
[FLINK-29023][docs][table] Updating Jar statement document

### PR DESCRIPTION
**Pull request by lsyldliu**
## What is the purpose of the change

Updating Jar statement document

## Brief change log

Updating Jar statement document

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (yes)
  - If yes, how is the feature documented? (docs)

### CODE REVIEW

**Comment by lsyldliu** [comment]
@wuchong

File: docs/content/docs/dev/table/sql/jar.md [hunk 1]
```diff
@@ -68,22 +70,24 @@ Flink SQL> REMOVE JAR '/path/hello.jar';
...
-## REMOVE JAR
+## SHOW JARS
 
 ```sql
-REMOVE JAR '<path_to_filename>.jar'
+SHOW JARS
 ```
 
-Currently it only supports to remove the jar that is added by the [`ADD JAR`](#add-jar) statements.
+Show all added jars in the session classloader which are added by [`ADD JAR`](#add-jar) and [`USING JAR`]({{< ref "docs/dev/table/sql/create" >}}#create-function) statements.
```
Comment (wuchong) [inline docs/content/docs/dev/table/sql/jar.md:81]: Why `SHOW JARS` also shows jars in `CREATE FUNCTION USING JAR`? The semantics of `SHOW/ADD/REMOVE JARS` seem not consistent to me. From my understanding, `SHOW JARS` should show the jars added to the session, but `CREATE FUNCTION USING JAR` doesn't affect the current session and functions with using jar referenced by queries are bound to the queries not sessions.

File: docs/content/docs/dev/table/sql/jar.md [hunk 2]
```diff
@@ -68,22 +70,24 @@ Flink SQL> REMOVE JAR '/path/hello.jar';
...
-## SHOW JARS
+## REMOVE JAR
 
 ```sql
-SHOW JARS
+REMOVE JAR '<path_to_filename>.jar'
 ```
 
-Show all added jars in the session classloader which are added by [`ADD JAR`](#add-jar) statements.
+Currently it supports to remove the jar that is added by the [`ADD JAR`](#add-jar) and [`USING JAR`]({{< ref "docs/dev/table/sql/create" >}}#create-function) statements.
```
Comment (wuchong) [inline docs/content/docs/dev/table/sql/jar.md:89]: I'm a little confused that why and how does REMOVE JAR remove USING JAR.

File: docs/content/docs/dev/table/sql/jar.md [hunk 3]
```diff
@@ -51,10 +48,15 @@ The following examples show how to run `JAR` statements in [SQL CLI]({{< ref "do
...
-[INFO] The specified jar is added into session classloader.
+[INFO] Execute statement succeed.
+
+Flink SQL> ADD JAR 'hdfs:///udf/common-udf.jar';
+[INFO] Execute statement succeed.
 
 Flink SQL> SHOW JARS;
+jars
 /path/hello.jar
+hdfs:///udf/common-udf.jar
```
Comment (wuchong) [inline docs/content/docs/dev/table/sql/jar.md:59]: Is this the style printed by the SQL CLI? Is there a table format around the result?

Comment (lsyldliu) [reply docs/content/docs/dev/table/sql/jar.md:59]: Yes, you are right

File: docs/content/docs/dev/table/sql/jar.md (hunk 1 above)
Comment (lsyldliu) [reply docs/content/docs/dev/table/sql/jar.md:81]: After reading spark [add jar](https://spark.apache.org/docs/latest/sql-ref-syntax-aux-resource-mgmt-add-jar.html) and [list jar](https://spark.apache.org/docs/latest/sql-ref-syntax-aux-resource-mgmt-list-jar.html) document, I think you are right, the show jars only list the jar which is added by add jar statement, this seems the semantic is consistent.
However, I've operated the spark sql and hive cli, I found that the list jars statement also shows the jars which are added `Create function ... using jar` statement, the document and actual behavior is not consistent. So regarding the document, I think we can follow the spark.

**Comment by lsyldliu** [comment]
@wuchong Thanks for review, I've updated it.

**Comment by lsyldliu** [comment]
@flinkbot run azure

**Review by wuchong — APPROVED** [review]
LGTM. I added a commit to update some words. 
I will merge it once CI is passed if you don't have concerns.

**Comment by wuchong** [comment]
The failed cases are tracked by FLINK-29315 and FLINK-29093.

**Pull request merged**